server/auth/types/saml/routes.ts

```diff
@@ -56,7 +56,7 @@ export class SamlAuthRoutes {
           const samlHeader = await this.securityClient.getSamlHeader(request);
           const cookie: SecuritySessionCookie = {
             saml: {
-              nextUrl: request.query.nextUrl,
+              nextUrl: `${this.coreSetup.http.basePath.serverBasePath}${request.query.nextUrl}`,
```

Comment:
Similar concern here, is there a case `serverBasePath` has no ending slash and  `nextUrl` has no leading slash?

Reply:
nextUrl will always start with slash

Reply:
I did a test on a server with basepath `/_plugin/kibana`. When I refresh after session expired on a GetStarted page, it redirected me to login page with `?nextUrl=%2F_plugin%2Fkibana%2Fapp%2Fopendistro_security#/getstarted` (while '%2F' is encoded '/'). My concern is that in this case it seems the basepath is doubled.

Reply:
makes sense, I should make the change in the next URL generation part